client/js/src/client.ts

```diff
@@ -1042,7 +1041,14 @@ export function api_factory(
 							)
 						);
 					} else if (event_callbacks[event_id]) {
-						await event_callbacks[event_id](_data);
+						if (_data.msg === "process_completed") {
+							unclosed_events.delete(event_id);
+							if (unclosed_events.size === 0) {
+								close_stream();
+							}
+						}
+						let fn = event_callbacks[event_id];
+						window.setTimeout(fn, 0, _data);
```

Comment:
Could we add a link to the PR in a comment here, just to provide context since this is slightly strange?

Reply:
Agreed. I think it would be easy to change this in a refactor inadvertently. How can we test this?

Reply:
Can't think of any way to test this with our existing framework - the DOM already *does* update, it's the browser that skips the intermediate repaints, which we can't test for with playwright